pkg/operator/operator.go

```diff
@@ -47,7 +49,11 @@ type Operator struct {
 	osClient      osclientset.Interface
 	eventRecorder record.EventRecorder
 
-	syncHandler func(ic string) error
+	syncHandler           func(ic string) error
+	imageGetterFunc       imageGetterFuncType
+	statusProgressingFunc func() error
```

Comment:
No need to mock `optr.statusProgressing` given you can have fake clients return expected objects. Please, give fake client `osconfigv1.ClusterOperator` object with pre-populated testing values.

Reply:
Implemented here: https://github.com/openshift/machine-api-operator/pull/304